iree/turbine/kernel/wave/codegen/read_write.py

```diff
@@ -583,28 +641,67 @@ def _create_vec_write(
         data, offset_th = _linearize_memref(
             mem, start_indices_wg, start_indices_th, strides
         )
-        offset_th = vector_d.splat(offsets_vec.type, offset_th)
-        offsets_vec = arith_d.addi(offsets_vec, offset_th)
-        if mask is not None:
+
+        if splatted_masked:

```

Comment:
The comment below says we can use vector buffer ops but I don't see any vector buffer ops. What did you mean by vector buffer ops?

Reply:
That means we can pass vector values directly to `raw_buffer_store`/ use vector `load_type` for `raw_buffer_load`

Reply:
makes sense, thanks